shop/src/main/java/ro/msg/learning/shop/controller/CustomerController.java

```diff
@@ -0,0 +1,26 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.learning.shop.dto.CustomerDto;
+import ro.msg.learning.shop.service.CustomerService;
+
+@RequestMapping("/customer")
+@Validated
```

Comment:
Not sure if the `@Validated` is used correctly here. Please check if is really needed in our case now and if yes you have to read how to use it.